pkg/kubelet/kubelet.go

```diff
@@ -729,7 +732,8 @@ func NewMainKubelet(kubeCfg *componentconfig.KubeletConfiguration, kubeDeps *Kub
 		klet.containerRuntime,
 		kubeDeps.Mounter,
 		klet.getPodsDir(),
-		kubeDeps.Recorder)
+		kubeDeps.Recorder,
+		kubeCfg.ExperimentalCheckMountBinariesType)
```

Comment:
Here is how you can implement the "put the pod in a failed state" functionality:
* In the volumeManager, introduce a new custom error, `NonRecoverableVolumeError`.
  * See `AlreadyExistsError` [in goroutinemap](https://github.com/kubernetes/kubernetes/blob/master/pkg/util/goroutinemap/goroutinemap.go#L183), for example.
* Have operation executor return the new `NonRecoverableVolumeError` instead of a standard `error` when the `CanMount()` check fails.
* In this file, modify `volumeManager.WaitForAttachAndMount` to handle the new `volumemanager.NonRecoverableVolumeError` and when it is detected, in addition to the usual error handling behavior, set the pod status to failed.
  * See `rejectPod()` method in this file for example.